lib/instrumentation/modules/aws-sdk/sns.js

```diff
@@ -0,0 +1,141 @@
+const constants = require('../../../constants')
+
+const TYPE = 'messaging'
+const SUBTYPE = 'sns'
+const ACTION = 'send'
+const ACCESS_POINT = 'accesspoint'
+const PHONE_NUMBER = '<PHONE_NUMBER>'
+
+function getRegionFromRequest (request) {
+  return request && request.service &&
+        request.service.config && request.service.config.region
+}
+
+function getSpanNameFromRequest (request) {
+  const topicName = getDestinationNameFromRequest(request)
+  return `SNS PUBLISH ${topicName}`
+}
+
+function getMessageContextFromRequest (request) {
+  return {
+    queue: {
+      name: getDestinationNameFromRequest(request)
+    }
+  }
+}
+
+function getAddressFromRequest (request) {
+  return request && request.service && request.service.endpoint &&
+    request.service.endpoint.hostname
+}
+
+function getPortFromRequest (request) {
+  return request && request.service && request.service.endpoint &&
+    request.service.endpoint.port
+}
+
+function getMessageDestinationContextFromRequest (request) {
+  return {
+    service: {
+      resource: `${SUBTYPE}/${getDestinationNameFromRequest(request)}`,
+      type: TYPE,
+      name: SUBTYPE,
+      address: getAddressFromRequest(request),
+      port: getPortFromRequest(request)
```

Comment:
Sorry, I had said `context.destination.service.{address,port}`, but it is [actually](https://github.com/elastic/apm/blob/master/specs/agents/tracing-spans-destination.md#destination-fields) `context.destination.{address,port}` so these two fields should move up a level.

Reply:
✅